File: backend/app/schemas/ai.py
from pydantic import BaseModel


class HighlightsRequest(BaseModel):
    media_id: str
    sensitivity: float = 0.5  # 0.0 (fewer) to 1.0 (more highlights)
    min_duration: float = 3.0  # minimum highlight duration in seconds


class ChaptersRequest(BaseModel):
    media_id: str
    min_chapter_duration: float = 30.0  # minimum chapter length in seconds


class SummaryRequest(BaseModel):
    media_id: str
    max_sentences: int = 5

File: frontend/src/types/ai.ts
export interface HighlightsRequest {
  media_id: string
  sensitivity?: number
  min_duration?: number
}

export interface HighlightSegment {
  start_ms: number
  end_ms: number
  duration_ms: number
  energy: number
}

export interface ChaptersRequest {
  media_id: string
  min_chapter_duration?: number
}

export interface Chapter {
  index: number
  start_ms: number
  end_ms: number
  title: string
}

export interface SummaryRequest {
  media_id: string
  max_sentences?: number
}
